builder/builder-source-git.c

```diff
@@ -417,75 +401,80 @@ builder_source_git_download (BuilderSource  *source,
 static gboolean
 git_extract_submodule (const char     *repo_location,
                        GFile          *checkout_dir,
+                       const char     *revision,
                        BuilderContext *context,
                        GError        **error)
 {
-  g_autofree char *submodule_status = NULL;
-
-  if (!git (checkout_dir, &submodule_status, error,
-            "submodule", "status", NULL))
-    return FALSE;
+  g_autoptr(GKeyFile) key_file = g_key_file_new ();
+  g_autofree gchar *submodule_data = NULL;
+  g_autofree gchar **submodules = NULL;
+  g_autofree gchar *gitmodules = g_strconcat (revision, ":.gitmodules", NULL);
+  gsize num_submodules;
 
-  if (submodule_status)
+  if (git (checkout_dir, &submodule_data, NULL, "show", gitmodules, NULL))
     {
```

Comment:
How does this handle the case where there is not submodules? I.e. if git returns a nonzero exist status?